packages/apollo-link-state/src/utils.ts

```diff
@@ -134,12 +134,26 @@ function selectionSetFromObj(obj) {
 export function addWriteDataToCache(cache: ApolloCacheClient) {
   cache.writeData = ({ id, data }: WriteDataArgs) => {
     if (id) {
-      cache.writeFragment({
-        fragment: fragmentFromPojo(data),
+      // Since we can't use fragments without having a typename in the store,
+      // we need to make sure we have one.
+      // To avoid overwriting an existing typename, we need to read it out first
+      // and generate a fake one if none exists.
+      const typenameResult: any = cache.read({
+        rootId: id,
+        optimistic: false,
+        query: justTypenameQuery,
+      });
+
+      // tslint:disable-next-line
```

Comment:
😬 let's just delete tslint when we get a chance

Reply:
YES PLS 🙌